nilearn/__init__.py

```diff
@@ -25,3 +26,17 @@
 # structures
 # This  is used in nilearn._utils.cache_mixin
 check_cache_version = True
+
+# These are warnings that we may trigger, but that we can do nothing about.
+# Hide them from users; warnings should be triggered for things that they
+# can act on only!
+import warnings
+
+# We know this, no need to expose our users to it.
+warnings.filterwarnings('ignore', 'check_cv will return indices instead of boolean masks', DeprecationWarning)
+warnings.filterwarnings('ignore', 'The compiler package is deprecated and removed in Python 3.x.', DeprecationWarning)
```

Comment:
>Is the regexp at the end of the message desired? ('.' in regexp matches anything).

For me, not particularly desired or undesired.  With a match of this length and specificity, I didn't feel the need to be careful about the presence of `.`.  I would be happy to remove it if you'd like.